src/storage/kv/cursor.rs

```diff
@@ -119,6 +125,7 @@ impl<I: Iterator> Cursor<I> {
     /// around `key`, otherwise you should use `seek` instead.
     pub fn near_seek(&mut self, key: &Key, statistics: &mut CfStatistics) -> Result<bool> {
         assert_ne!(self.scan_mode, ScanMode::Backward);
+        self.in_bound = true;
```

Comment:
I prefer to add unit test cases for calling `near_seek` directly because of the logics added here.